dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboCodec.java

```diff
@@ -152,12 +154,31 @@ protected Object decodeBody(Channel channel, InputStream is, byte[] header) thro
                 } else {
                     req = new HeartBeatRequest(id);
                     DecodeableRpcInvocation inv;
+                    String exPs = System.getProperty(EXCEPTION_PROCESSOR_KEY);
+                    boolean existExpHandler = false;
+                    ExtensionLoader<ExceptionProcessor> extensionLoader;
+                    ExceptionProcessor expProcessor = null;
+                    if (StringUtils.isNotBlank(exPs)) {
+                        existExpHandler = true;
+                        extensionLoader = channel.getUrl().getOrDefaultFrameworkModel().getExtensionLoader(ExceptionProcessor.class);
+                        expProcessor = extensionLoader.getOrDefaultExtension(exPs);
+                    }
+
```

Comment:
Cached in constructor to prevent performance hit

Reply:
done